### PR TITLE
Fix MXNet data locality

### PR DESCRIPTION
**Pull request by hkvision**
Previously using kv.rank to get the data doesn't mean the worker ips are sorted.

Tested on YARN cluster.

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/3584/

File: pyzoo/zoo/orca/learn/mxnet/mxnet_trainer.py [hunk 1]
```diff
@@ -124,7 +124,23 @@ def __init__(self, config, train_data, model_creator,
 
         if isinstance(self.train_data, RayXShards):
             self.workers = self.train_data.colocate_actors(self.workers)
+            train_data_list = self.train_data.get_partitions()
+            if self.test_data:
+                test_data_list = self.test_data.get_partitions()
+            else:
+                test_data_list = [self.test_data] * self.num_workers
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/mxnet/mxnet_trainer.py:131]: [None] * self.num_workers

Comment (hkvision) [reply pyzoo/zoo/orca/learn/mxnet/mxnet_trainer.py]: Updated.

**Comment by hkvision** [comment]
http://10.239.47.210:18888/job/ZOO-PR-Validation/3587/

**Pull request merged**